python/lsst/ts/watcher/alarm.py

```diff
@@ -405,10 +437,92 @@ def assert_equal(self, other, ignore_attrs=()):
             error_str = ", ".join(diffs)
             raise AssertionError(error_str)
 
+    async def assert_next_severity(
+        self,
+        expected_severity,
+        check_empty=True,
+        flush=False,
+        timeout=DEFAULT_NEXT_SEVERITY_TIMEOUT,
+    ):
+        """Wait for and check the next severity.
+
+        Only intended for tests.
+        In order to call this you must first call `init_severity_queue`
+        (once) to set up a severity queue.
+
+        Parameters
+        ----------
+        expected_severity : AlarmSeverity
+            The expected severity.
+        check_empty : bool, optional
+            If true (the default): check that the severity queue is empty,
+            after getting the severity.
+        flush : bool, optional
+            If true (not the default): flush all existing values
+            from the queue, then wait for the next severity.
+            This is useful for polling alarms.
+        timeout : float, optional
+            Maximum time to wait (seconds)
+
+        Raises
+        ------
+        AssertionError
+            If the severity is not as expected, or if ``check_empty`` true
+            and there are additional queued severities.
+        asyncio.TimeoutError
+            If no new severity is seen in time.
+        RuntimeError
+            If you never called `init_severity_queue`.
+
+        Notes
+        -----
+        Here is the typical way to use this method:
+        * Create a rule
+        * Call `rule.alarm.init_severity_queue()`
+        * Write SAL messages that are expected to change the alarm severity.
+        * After writing each such message, call::
+
+            await rule.alarm.assert_next_severity(expected_severity)
+        """
+        if self.severity_queue is None:
+            raise RuntimeError(
+                "No severity queue; you must call init_severity_queue "
+                "(once) before calling assert_next_severity"
+            )
+        if flush:
+            while not self.severity_queue.empty():
+                self.severity_queue.get_nowait()
+        severity = await asyncio.wait_for(self.severity_queue.get(), timeout=timeout)
+        if check_empty:
+            extra_severities = [
+                self.severity_queue.get_nowait()
+                for i in range(self.severity_queue.qsize())
+            ]
+            if extra_severities:
+                raise AssertionError(
+                    f"severity_queue was not empty; it contained {extra_severities}"
+                )
+        if severity != expected_severity:
+            raise AssertionError(
+                f"severity={severity!r} != expected_severity{expected_severity!r}"
+            )
+
+    def init_severity_queue(self):
+        """Initialize the severity queue.
+
+        Call this once before calling `assert_next_severity`.
```

Comment:
What happens if you call it more than once? My guess is that you lose any previously stored severity items. Is there a way to check and verify that this hasn't happened before initializing the Queue?

Reply:
Yes it replaces the queue. Not a common use case but I'll add some text.